src/Umbraco.Web.Common/RuntimeMinification/SmidgeOptionsSetup.cs

```diff
@@ -12,11 +13,25 @@ public SmidgeOptionsSetup(IOptions<RuntimeMinificationSettings> runtimeMinificat
         => _runtimeMinificatinoSettings = runtimeMinificatinoSettings;
 
     /// <summary>
-    ///     Configures Smidge to use in-memory caching if configured that way or if certain cache busters are used
+    ///     Configures Smidge to use in-memory caching if configured that way or if certain cache busters are used. As well as the cache buster type.
     /// </summary>
     /// <param name="options"></param>
     public void Configure(SmidgeOptions options)
-        => options.CacheOptions.UseInMemoryCache = _runtimeMinificatinoSettings.Value.UseInMemoryCache ||
+    {
+        options.CacheOptions.UseInMemoryCache = _runtimeMinificatinoSettings.Value.UseInMemoryCache ||
                                                    _runtimeMinificatinoSettings.Value.CacheBuster ==
                                                    RuntimeMinificationCacheBuster.Timestamp;
+
+
+        Type cacheBusterType = _runtimeMinificatinoSettings.Value.CacheBuster switch
+        {
+            RuntimeMinificationCacheBuster.AppDomain => typeof(AppDomainLifetimeCacheBuster),
+            RuntimeMinificationCacheBuster.Version => typeof(UmbracoSmidgeConfigCacheBuster),
+            RuntimeMinificationCacheBuster.Timestamp => typeof(TimestampCacheBuster),
+            _ => throw new NotImplementedException(),
```

Comment:
Could we use `ArgumentOutOfRangeException` here, along with a message indicating the issue?  I can see you copied this from `SmidgeRuntimeMinifier`, but would good to make the same update there.